Doc(eslint-plugin-fiori-tools): update DataField rule documentation with semantic link information (#4506)
SAP/open-ux-tools: packages/eslint-plugin-fiori-tools/docs/rules/sap-no-data-field-intent-based-navigation.md

</Annotation>
```

> ### Correct: Semantic Link or Smart Link Navigation Is Used (via `Common.SemanticObject` on Entity Property)

Reviewer: ```suggestion
### Correct: Semantic Link or Smart Link Navigation Is Used Using `Common.SemanticObject` on the Entity Property
```

> Annotate the entity property (e.g., `CustomerId`) with `Common.SemanticObject` before using it in `UI.LineItem` or `UI.FieldGroup`. SAP Fiori Elements automatically detects this and renders the field as a Semantic Link with Smart Link popover support.

Reviewer: ```suggestion
Annotate the entity property, for example,`CustomerId`, with the `Common.SemanticObject` annotation before using it in a `UI.LineItem` or `UI.FieldGroup`. SAP Fiori elements automatically detects this and renders the field as a semantic link with support for a smart link popover.
```

#### Step 1: Annotate the Entity Property:

```xml
<!-- Applied to the actual OData property, NOT inside DataField -->
<Annotation Term="Common.SemanticObject" String="Customer" Target="YourEntityType/CustomerId"/>
```

> #### Step 2: Reference the Property in UI Annotations (no special navigation annotation needed):

Reviewer: ```suggestion
#### Step 2: Reference the Property in Your UI Annotations:

Note: No navigation annotation is needed.
```

> For `UI.LineItem` (Table Column):

Reviewer: ```suggestion
For a table column using the `UI.LineItem` annotation:
```

```xml
<Annotation Term="UI.LineItem">
    <Collection>
        <Record Type="UI.DataField">
            <PropertyValue Property="Value" Path="CustomerId"/>
            <PropertyValue Property="Label" String="Customer"/>
            <!-- Semantic Link rendered automatically due to Common.SemanticObject on CustomerId property -->
        </Record>
    </Collection>
</Annotation>
```

> For `UI.FieldGroup` (Form Field):

Reviewer: ```suggestion
For a form field using the `UI.FieldGroup` annotation:
```

```xml
<Annotation Term="UI.FieldGroup" Qualifier="GeneralInformation">
    <Record Type="UI.FieldGroupType">
        <PropertyValue Property="Data">
            <Collection>
                <Record Type="UI.DataField">
                    <PropertyValue Property="Value" Path="CustomerId"/>
                    <PropertyValue Property="Label" String="Customer"/>
                    <!-- Semantic Link rendered automatically due to Common.SemanticObject on CustomerId property -->
                </Record>
            </Collection>
        </PropertyValue>
    </Record>
</Annotation>
```

## Bug Report

If you encounter an issue with this rule, please open a [GitHub issue](https://github.com/SAP/open-ux-tools/issues).

## Further Reading

> - [Semantic Link implementation in Fiori Elements](https://ui5.sap.com/#/topic/c18ada4bc56e427a9a2df2d1898f28a5)

Reviewer: ```suggestion
- [Semantic Link Implementation in SAP Fiori Elements](https://ui5.sap.com/#/topic/c18ada4bc56e427a9a2df2d1898f28a5)
```

> - [Smart Link / Navigation Intents](https://ui5.sap.com/#/topic/d782acf8bfd74107ad6a04f0361c5f62)

Reviewer: ```suggestion
- [Smart Link and Navigation Intents](https://ui5.sap.com/#/topic/d782acf8bfd74107ad6a04f0361c5f62)
```

</Annotation>
```

### Correct: Semantic Link or Smart Link Navigation Is Used Using `Common.SemanticObject` on the Entity Property

> Annotate the entity property, for example,`CustomerId`, with the `Common.SemanticObject` annotation before using it in a `UI.LineItem` or `UI.FieldGroup`. SAP Fiori elements automatically detects this and renders the field as a semantic link with support for a smart link popover.

Reviewer 2: ```suggestion
Annotate the entity property, for example,`CustomerId`, with the `Common.SemanticObject` annotation before using it in a `UI.LineItem` or `UI.FieldGroup`. SAP Fiori elements automatically detects this and renders the field as a semantic link.
```

#### Step 1: Annotate the Entity Property:

```xml
<!-- Applied to the actual OData property, NOT inside DataField -->

> <Annotation Term="Common.SemanticObject" String="Customer" Target="YourEntityType/CustomerId"/>

Reviewer 3: Can we set Target on Annotation?
Shouldn't we write the following?

```
<Annotations Target="YourEntityType/CustomerId">
    <Annotation Term="Common.SemanticObject" String="Customer"/>
</Annotations>
```

Author: Hi @hitesh-parmar can you help with Marco's question?

Reviewer 2: I believe Marco is right, I don't remember an example where Term and Target are on the same level. I think annotation LSP should show an error/warning in this case.

Author: So do we need to update this snippet?

Author: It's replaced.
